google-cloud-firestore/lib/google/cloud/firestore/transaction.rb

```diff
@@ -0,0 +1,640 @@
+# Copyright 2017 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+require "google/cloud/firestore/collection_reference"
+require "google/cloud/firestore/document_reference"
+require "google/cloud/firestore/document_snapshot"
+require "google/cloud/firestore/commit_response"
+require "google/cloud/firestore/convert"
+
+module Google
+  module Cloud
+    module Firestore
+      ##
+      # # Transaction
+      #
+      # A transaction in Cloud Firestore is a set of reads and writes that
+      # execute atomically at a single logical point in time.
+      #
+      # All changes are accumulated in memory until the block passed to
+      # {Database#transaction} completes. Transactions will be automatically
+      # retried when possible. See {Database#transaction}.
```

Comment:
Can you say something like `when documents change before committing the transaction`.

Reply:
Updated.